File: core/compiler/objc/UMCore.m
#import "UMCore.h"

@implementation UMCore
- (instancetype)initWithJson:(NSString *)json {
	return [super init];
}

- (NSString *)encodeToJson {
	return @"";
}
@end



@implementation UMCoreBasicUser
- (instancetype)init {
	if (self = [super init]) {
		_name = @"";
		_age = 0;
		
	}
	return self;
}

- (instancetype)initWithJson:(NSDictionary *)json {
	if (self = [self init]) {
		id attribute = nil;

		attribute = [json objectForKey:@"name"];
		if (attribute) {
			_name = attribute;
			
		}
		else {
			_name = @"";
			}
	attribute = [json objectForKey:@"age"];
		if (attribute) {
			_age = ((NSNumber *)attribute).integerValue;
			
		}
		else {
			_age = 0;
			}
	
	}
	return self;
}

- (NSDictionary *)encodeToJson {
	NSMutableDictionary *json = [NSMutableDictionary dictionary];

	[json setObject:_name forKey:@"name"];
	[json setObject:@(_age) forKey:@"age"];
	
	return json;
}


- (void)setName:(NSString *)name {
	
	_name = name;
}

- (void)setAge:(NSInteger )age {
	
	_age = age;
}

@end



@implementation UMCoreHttpGetUserRequest
- (instancetype)init {
	if (self = [super init]) {
		
	}
	return self;
}

- (instancetype)initWithJson:(NSDictionary *)json {
	if (self = [self init]) {
		id attribute = nil;

		
	}
	return self;
}

- (NSDictionary *)encodeToJson {
	NSMutableDictionary *json = [NSMutableDictionary dictionary];

	
	return json;
}


@end


@implementation UMCoreHttpGetUserResponse
- (instancetype)init {
	if (self = [super init]) {
		_user = [UMCoreBasicUser new];
		
	}
	return self;
}

- (instancetype)initWithJson:(NSDictionary *)json {
	if (self = [self init]) {
		id attribute = nil;

		attribute = [json objectForKey:@"user"];
		if (attribute) {
			_user = [[UMCoreBasicUser alloc] initWithJson:attribute];
			
		}
		else {
			_user = [UMCoreBasicUser new];
			}
	
	}
	return self;
}

- (NSDictionary *)encodeToJson {
	NSMutableDictionary *json = [NSMutableDictionary dictionary];

	[json setObject:[_user encodeToJson] forKey:@"user"];
	
	return json;
}


- (void)setUser:(UMCoreBasicUser *)user {
	
	_user = user;
}

@end


@implementation UMCoreHttpSetUserRequest
- (instancetype)init {
	if (self = [super init]) {
		_user = [UMCoreBasicUser new];
		
	}
	return self;
}

- (instancetype)initWithJson:(NSDictionary *)json {
	if (self = [self init]) {
		id attribute = nil;

		attribute = [json objectForKey:@"user"];
		if (attribute) {
			_user = [[UMCoreBasicUser alloc] initWithJson:attribute];
			
		}
		else {
			_user = [UMCoreBasicUser new];
			}
	
	}
	return self;
}

- (NSDictionary *)encodeToJson {
	NSMutableDictionary *json = [NSMutableDictionary dictionary];

	[json setObject:[_user encodeToJson] forKey:@"user"];
	
	return json;
}


- (void)setUser:(UMCoreBasicUser *)user {
	
	_user = user;
}

@end


@implementation UMCoreHttpSetUserResponse
- (instancetype)init {
	if (self = [super init]) {
		
	}
	return self;
}

- (instancetype)initWithJson:(NSDictionary *)json {
	if (self = [self init]) {
		id attribute = nil;

		
	}
	return self;
}

- (NSDictionary *)encodeToJson {
	NSMutableDictionary *json = [NSMutableDictionary dictionary];

	
	return json;
}


@end

File: core/compiler/php/UMCore.php
<?php
namespace UMCore {
	abstract class Model {
		abstract public function encodeToJson();
	}
}


namespace UMCore\Basic {
	class User extends \UMCore\Model {
		
		private $_name;
		private $_age;
		

		public function __construct($json = null) {
			if (isset($json) && (isset($json['name']))) {
				$this->_name = (string)($json['name']);
				
			}
			else {
				$this->_name = "";
				
			}
			if (isset($json) && (isset($json['age']))) {
				$this->_age = (int)($json['age']);
				
			}
			else {
				$this->_age = 0;
				
			}
			
		}

		public function encodeToJson() {
			$json = array();
			$json['name'] = $this->_name;
			$json['age'] = $this->_age;
			
			return $json;
		}
		

		public function getName() {
			return $this->_name;
		}

		public function setName($name) {
			
			$this->_name = $name;
		}	
		

		public function getAge() {
			return $this->_age;
		}

		public function setAge($age) {
			
			$this->_age = $age;
		}	
		
	}
	
}
namespace UMCore\Http {
	class GetUserRequest extends \UMCore\Model {
		
		

		public function __construct($json = null) {
			
		}

		public function encodeToJson() {
			$json = array();
			
			return $json;
		}
		
	}
	class GetUserResponse extends \UMCore\Model {
		
		private $_user;
		

		public function __construct($json = null) {
			if (isset($json) && (isset($json['user']))) {
				$this->_user = new \UMCore\Basic\User($json['user']);
				
			}
			else {
				$this->_user = new \UMCore\Basic\User();
				
			}
			
		}

		public function encodeToJson() {
			$json = array();
			$json['user'] = $this->_user->encodeToJson();
			
			return $json;
		}
		

		public function getUser() {
			return $this->_user;
		}

		public function setUser($user) {
			
			$this->_user = $user;
		}	
		
	}
	class SetUserRequest extends \UMCore\Model {
		
		private $_user;
		

		public function __construct($json = null) {
			if (isset($json) && (isset($json['user']))) {
				$this->_user = new \UMCore\Basic\User($json['user']);
				
			}
			else {
				$this->_user = new \UMCore\Basic\User();
				
			}
			
		}

		public function encodeToJson() {
			$json = array();
			$json['user'] = $this->_user->encodeToJson();
			
			return $json;
		}
		

		public function getUser() {
			return $this->_user;
		}

		public function setUser($user) {
			
			$this->_user = $user;
		}	
		
	}
	class SetUserResponse extends \UMCore\Model {
		
		

		public function __construct($json = null) {
			
		}

		public function encodeToJson() {
			$json = array();
			
			return $json;
		}
		
	}
	
}
